packages/react-components/src/components/chart/trendCursor/seriesYMinMax/useHandleSeriesYMinMax.ts

```diff
@@ -0,0 +1,54 @@
+import { Dispatch, MutableRefObject, SetStateAction, useEffect, useRef } from 'react';
+import { InternalGraphicComponentGroupOption } from '../types';
+import { YAXisOption } from 'echarts/types/dist/shared';
+import { Visualization, YAxisLegendOption } from '../../types';
+import { calculateSeriesMakers } from '../calculations/calculateSeriesMakers';
+import { ECharts, SeriesOption } from 'echarts';
+import { updateTrendCursorLineMarkers } from '../getTrendCursor/components/markers';
+
+export const useHandleSeriesYMinMax = ({
+  yAxisOptions,
+  graphic,
+  setGraphic,
+  chartRef,
+  visualization,
+  series,
+}: {
+  yAxisOptions: {
+    yAxis: YAXisOption[];
+    yMins: YAxisLegendOption[];
+    yMaxs: YAxisLegendOption[];
+  };
+  graphic: InternalGraphicComponentGroupOption[];
+  setGraphic: Dispatch<SetStateAction<InternalGraphicComponentGroupOption[]>>;
+  chartRef: MutableRefObject<ECharts | null>;
+  visualization: Visualization;
+  series: SeriesOption[];
+}) => {
+  const seriesRef = useRef(series);
+  const visualizationRef = useRef(visualization);
+  const graphicRef = useRef(graphic);
+  const yAxisOptionsString = JSON.stringify(yAxisOptions);
+  const seriesString = JSON.stringify(series);
+
+  useEffect(() => {
+    graphicRef.current = graphic;
+    visualizationRef.current = visualization;
+    seriesRef.current = series;
+  }, [graphic, visualization, series]);
+
+  useEffect(() => {
+    const newG = graphicRef.current.map((g) => {
+      const { trendCursorsSeriesMakersInPixels, trendCursorsSeriesMakersValue } = calculateSeriesMakers(
+        seriesRef.current,
+        g.timestampInMs,
+        chartRef,
+        visualizationRef.current
+      );
+      g.yAxisMarkerValue = trendCursorsSeriesMakersValue;
+      g.children = updateTrendCursorLineMarkers(g.children, trendCursorsSeriesMakersInPixels);
+      return g;
+    });
+    setGraphic([...newG]);
+  }, [yAxisOptionsString, setGraphic, chartRef, seriesString]);
```

Comment:
Why are we using useEffect to update the state? Is there no event occurring we can call attach a callback to?

Reply:
nope, the function inside the useEffect needs to run when the user/api updates the ymin or ymax or series... the change in value itself is an event within the component,,, we can do use the prev reference to achieve this probably, but useEffect made the code look clean.... open for suggestions